lib/operations/re_index.js

```diff
@@ -1,28 +1,34 @@
 'use strict';
 
-const CommandOperation = require('./command');
-const handleCallback = require('../utils').handleCallback;
+const Aspect = require('./operation').Aspect;
+const defineAspects = require('./operation').defineAspects;
+const CommandOperationV2 = require('./command_v2');
+const serverType = require('../core/sdam/common').serverType;
+const ServerType = require('../core/sdam/common').ServerType;
+const MongoError = require('../core').MongoError;
 
-class ReIndexOperation extends CommandOperation {
+class ReIndexOperation extends CommandOperationV2 {
   constructor(collection, options) {
-    super(collection.s.db, options, collection);
+    super(collection.s.db, options);
+    this.collection = collection;
+    this.command = { reIndex: collection.collectionName };
```

Comment:
remember the guiding principle of operations v2 is: 
- save all things needed for building the command in the constructor. Try to make an instance of this class represent the [command](https://docs.huihoo.com/mongodb/3.4/reference/command/reIndex/index.html) as much as possible. In this case its a really limited command, it might just be that `this.collection` is saved off. 
- actually build the command in `execute`